aws/resource_aws_vpn_gateway.go

```diff
@@ -178,6 +178,20 @@ func resourceAwsVpnGatewayDelete(d *schema.ResourceData, meta interface{}) error
 	})
 }
 
+func resourceAwsVpnGatewayCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	if diff.Id() == "" {
+		// Only validate Amazon-side ASN for new resources.
+		// Imported resources may have legacy 7224 or 9059 ASNs.
+		if v, ok := diff.GetOk("amazon_side_asn"); ok {
```

Comment:
If I'm understanding this correctly, this logic doesn't prevent the validation issue after creation where `amazon_side_asn = 7224` is present in the configuration. Since it is technically valid, I think instead of adding all this workaround code that we should simply loosen our plan time validation to accept the two legacy values.

Reply:
Yes we can do that.